### PR TITLE
SALTO-3080 Add html parser for urls with references

### PR DESCRIPTION
**Pull request by adfineberg**
Currently just logging the references extracted from HTML url tags.

---

_Additional context for reviewer_

---
_Release Notes_: 
_Replace me with a short sentence that describes the effect of this change on Salto users_

---
_User Notifications_: 
_Replace me with a short sentence that describes changes that will appear in NaCls and are not caused by user actions (e.g. a new annotation, field values that are converted to references, etc). Hidden changes should not be listed._

### CODE REVIEW

File: packages/zendesk-adapter/src/filters/guide_theme.ts [hunk 1]
```diff
@@ -66,15 +68,49 @@ export type ThemeDirectory = {
...
+        idsToElements[ref.value] !== undefined
+          ? { value: new ReferenceExpression(idsToElements[ref.value].elemID, idsToElements[ref.value]), loc: ref.loc }
+          : ref,
       )
-      if (templateParts.length > 0) {
-        log.info('Found the following references in the theme: %o', templateParts)
+      if (handlebarReferences.length > 0) {
+        log.info('Found the following references in file %s in the theme: %o', filePath, handlebarReferences)
+      }
+      const { urls } = parseHtmlPotentialReferences(content, {
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/filters/guide_theme.ts:93]: why do you check for urls inside hbs file?

File: packages/zendesk-adapter/src/filters/template_engines/html_parser.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  return {
+    urls: urls.map(url => ({
+      value: parseUrlPotentialReferencesFromString(url.value, {
+        urlBrandInstance,
+        instancesById,
+        enableMissingReferences,
+      }),
+      loc: url.loc,
+    })),
+    scripts,
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/filters/template_engines/html_parser.ts:137]: do you plan to later look for potential references insude the scripts as well?

File: packages/zendesk-adapter/test/filters/template_engines/html_url_extractor.test.ts
Comment (edenhassid) [inline packages/zendesk-adapter/test/filters/template_engines/html_url_extractor.test.ts:1]: can you add a test for `parseHtmlPotentialReferences`?

**Review by edenhassid — COMMENTED** [review]
looks good! 🔥  added some questions and a comment :)

File: packages/zendesk-adapter/src/filters/template_engines/html_parser.ts (hunk 2 above)
Comment (adfineberg) [reply packages/zendesk-adapter/src/filters/template_engines/html_parser.ts:137]: yeah, that's the next PR - that handles the JS content (both files and script tags)

File: packages/zendesk-adapter/src/filters/guide_theme.ts (hunk 1 above)
Comment (adfineberg) [reply packages/zendesk-adapter/src/filters/guide_theme.ts:93]: hbs is an html file that can also have the Handlebar templates. So, they also add references there

**Review by edenhassid — APPROVED** [review]
LGTM! 🏆

**Pull request merged**